### PR TITLE
[KOGITO-4455] Adding Kogito Quarkus Workflows extension to generated …

### PR DESCRIPTION
**Pull request by ricardozanini**
…Quarkus projects

See: https://issues.redhat.com/browse/KOGITO-4455

In this PR we add the ability to add the Kogito Quarkus Workflows extension to the Quarkus projects built by the s2i tool. SpringBoot won't be supported at this time.

Depends on https://github.com/kiegroup/kogito-runtimes/pull/1153

Signed-off-by: Ricardo Zanini <zanini@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by ricardozanini** [comment]
On hold until the extension is published.

**Comment by ricardozanini** [comment]
The bats error is because we don´t have the extension published.

**Comment by ricardozanini** [comment]
Depends on https://github.com/kiegroup/kogito-images/pull/474

**Comment by ricardozanini** [comment]
Jenkins retest this

**Review by sutaakar — APPROVED** [review]

File: modules/kogito-s2i-core/added/s2i-core [hunk 1]
```diff
@@ -126,20 +126,41 @@ function build_kogito_app() {
...
         for item in *
         do
             if [ -d "${item}" ] && [ "${item}" == "$PROJECT_ARTIFACT_ID" ]; then
                 log_info "--> Skipping generated project ..."
             else
-                log_info "--> Coping resource ${item}"
+                log_info "--> Copying resource ${item}"
                 cp -Rv "${item}" $PROJECT_ARTIFACT_ID/src/main/resources
+
+                if [[ ${item} =~ .*\.sw\.(yaml|yml|json) && "${RUNTIME_TYPE}" == "${QUARKUS_RUNTIME_TYPE}" ]]; then
```
Comment (radtriste) [inline modules/kogito-s2i-core/added/s2i-core]: could we have also the handling of spring boot here ?
I mean display at least a message if not supported ;)

File: modules/kogito-s2i-core/added/s2i-core [hunk 2]
```diff
@@ -126,20 +126,41 @@ function build_kogito_app() {
...
 
         # move all project content into the current directory
         mv -v $PROJECT_ARTIFACT_ID/* .
 
+        # add extension if needed
+        if [ ${require_sw} = true ]; then
+            log_info "--> Adding Kogito Quarkus Workflows extension to the generated project."
+            
+            quarkus_version=$(echo '${quarkus.version}' | $MAVEN_HOME/bin/mvn help:evaluate -q -DforceStdout | grep -v '^[[]')
+            if [[ ! "$quarkus_version" =~ [0-9]{0,2}\.[0-9]{0,3} ]]; then
```
Comment (radtriste) [inline modules/kogito-s2i-core/added/s2i-core:159]: wouldn't be Quarkus version as M.m.µ ?

Comment (ricardozanini) [reply modules/kogito-s2i-core/added/s2i-core:159]: I just need to know that's a version, not a strict validation.

File: modules/kogito-s2i-core/added/s2i-core (hunk 1 above)
Comment (ricardozanini) [reply modules/kogito-s2i-core/added/s2i-core]: Sure

**Comment by ricardozanini** [comment]
Looks like we are using a very old version of the archetype:

> Archetype repository not defined. Using the one from [org.kie.kogito:kogito-quarkus-archetype:0.1.1] found in catalog remote

Thus, failing on getting the Quarkus version:

> ERROR --> Impossible to get Quarkus version from the generated Maven project (output: ' @ '), failing build

I'll run some tests.

**Review by radtriste — APPROVED** [review]

**Comment by ricardozanini** [comment]
Bats failed 'cause of the -X flag. I'll revert it after finding out the problem.

**Review by vaibhavjainwiz — APPROVED** [review]

**Comment by ricardozanini** [comment]
Documentation for the Archetypes Catalog and why we were using a super old version in our images:
https://maven.apache.org/archetype/maven-archetype-plugin/specification/archetype-catalog.html

@sutaakar @radtriste can you guys take a look again? I made some changes.

File: tests/features/kogito-builder.feature [hunk 3]
```diff
@@ -78,4 +78,28 @@ Feature: kogito-builder image tests
       | expected_phrase | Should the driver be suspended?                                                                  |
       | request_method  | POST                                                                                             |
       | content_type    | application/json                                                                                 |
-      | request_body    | {"Driver": {"Points": 2}, "Violation": {"Type": "speed","Actual Speed": 120,"Speed Limit": 100}} |
\ No newline at end of file
+      | request_body    | {"Driver": {"Points": 2}, "Violation": {"Type": "speed","Actual Speed": 120,"Speed Limit": 100}} |
+
+  @wip
```
Comment (sutaakar) [inline tests/features/kogito-builder.feature]:
```suggestion
```

**Review by sutaakar — APPROVED** [review]
Looks good to me, one leftover above

**Comment by ricardozanini** [comment]
Altghout we have this message in the logs:

```
2021-05-26 10:56:46,861 - root - ERROR - I0526 10:56:46.861635   15622 sti.go:711] [INFO] Archetype repository not defined. Using the one from [org.kie.kogito:kogito-quarkus-archetype:0.1.1] found in catalog remote
```

Running in DEBUG mode, we can see that the plugin actually downloads the right version:

```
2021-05-26 10:56:46,862 - root - ERROR - I0526 10:56:46.862172   15622 sti.go:711] [DEBUG] Not found archetype org.kie.kogito:kogito-quarkus-archetype:2.0.0-SNAPSHOT in cache
2021-05-26 10:56:46,867 - root - ERROR - I0526 10:56:46.866889   15622 sti.go:711] [DEBUG] Could not find metadata org.kie.kogito:kogito-quarkus-archetype:2.0.0-SNAPSHOT/maven-metadata.xml in local (/home/kogito/.m2/repository)

2021-05-26 10:56:47,117 - root - ERROR - I0526 10:56:47.117073   15622 sti.go:711] [DEBUG] Found archetype org.kie.kogito:kogito-quarkus-archetype:2.0.0-SNAPSHOT in cache: /home/kogito/.m2/repository/org/kie/kogito/kogito-quarkus-archetype/2.0.0-SNAPSHOT/kogito-quarkus-archetype-2.0.0-SNAPSHOT.jar
2021-05-26 10:56:47,117 - root - ERROR - I0526 10:56:47.117350   15622 sti.go:711] [DEBUG] checking fileset archetype status on /home/kogito/.m2/repository/org/kie/kogito/kogito-quarkus-archetype/2.0.0-SNAPSHOT/kogito-quarkus-archetype-2.0.0-SNAPSHOT.jar
2021-05-26 10:56:47,118 - root - ERROR - I0526 10:56:47.118272   15622 sti.go:711] [DEBUG] Searching for META-INF/maven/archetype-metadata.xml inside /home/kogito/.m2/repository/org/kie/kogito/kogito-quarkus-archetype/2.0.0-SNAPSHOT/kogito-quarkus-archetype-2.0.0-SNAPSHOT.jar
2021-05-26 10:56:47,119 - root - ERROR - I0526 10:56:47.119595   15622 sti.go:711] [DEBUG]   - META-INF/maven/archetype-metadata.xml
2021-05-26 10:56:47,120 - root - ERROR - I0526 10:56:47.119611   15622 sti.go:711] [DEBUG] Entry found
2021-05-26 10:56:47,120 - root - ERROR - I0526 10:56:47.120103   15622 sti.go:711] [DEBUG] Found archetype org.kie.kogito:kogito-quarkus-archetype:2.0.0-SNAPSHOT in cache: /home/kogito/.m2/repository/org/kie/kogito/kogito-quarkus-archetype/2.0.0-SNAPSHOT/kogito-quarkus-archetype-2.0.0-SNAPSHOT.jar
2021-05-26 10:56:47,120 - root - ERROR - I0526 10:56:47.120721   15622 sti.go:711] [DEBUG] Searching for META-INF/maven/archetype-metadata.xml inside /home/kogito/.m2/repository/org/kie/kogito/kogito-quarkus-archetype/2.0.0-SNAPSHOT/kogito-quarkus-archetype-2.0.0-SNAPSHOT.jar
2021-05-26 10:56:47,121 - root - ERROR - I0526 10:56:47.121079   15622 sti.go:711] [DEBUG]   - META-INF/maven/archetype-metadata.xml
2021-05-26 10:56:47,125 - root - ERROR - I0526 10:56:47.121101   15622 sti.go:711] [DEBUG] Entry found
```

I've removed the catalog file.

**Comment by ricardozanini** [comment]
@radtriste up to you to merge :)
I'd like to include it in 1.8.

**Review by radtriste — APPROVED** [review]
Nice ! :)

**Comment by radtriste** [comment]
@sutaakar @vaibhavjainwiz Good with latest changes ?

**Comment by sutaakar** [comment]
Fine for me

**Pull request merged**